### PR TITLE
[WIP] GSoC fixed-point compiler

### PR DESCRIPTION
**Pull request by qrqiuren**
This is the pull request for the implementation of the fixed-point compiler in GSoC 2017.

### CODE REVIEW

File: myhdl/_fixbv.py [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+            slc = intbv(self._val, _nrbits=self._nrbits)
+            return slc.__getitem__(key)
+
+    def __setitem__(self, key, val):
+        if isinstance(val, fixbv):
+            assert self._W._fwl == val._W._fwl
+            v = val._val
+        else:
+            v = val
+        # @todo: convert negative keys to the correct bit index
```
Comment (cfelton) [inline myhdl/_fixbv.py]: Need to add the round and alignment functions here, the majority of the work for this project is the automatic alignment and sizing.

File: myhdl/_fixbv.py [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+        return self._to_float() #self._fval
+
+    def __ord__(self):
+        return self._val
+
+    def __add__(self, other):
+        #if isinstance(other, _Signal):
+        #    other = other._val
+        if isinstance(other, fixbv):
+            assert self._W._fwl == other._W._fwl, \
```
Comment (cfelton) [inline myhdl/_fixbv.py]: The assert needs to be removed and the automatic alignment needs to occur.

File: myhdl/_fixbv.py [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+        retval = fixbv(0)[iW[:]]
+        retval._val = self._val + other._val
+        retval._handleBounds()
+        return retval
+
+    def __sub__(self, other):
+        #if isinstance(other, _Signal):
+        #    other = other._val
+        if isinstance(other, fixbv):
+            assert self._W._fwl == other._W._fwl, \
```
Comment (cfelton) [inline myhdl/_fixbv.py]: Need to remove the assert and implement the automatic alignment.

File: myhdl/_fixbv.py [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+        retval._val = self._val ** other
+        retval._handleBounds()
+        return retval
+
+    # all comparisons must be on aligned types
+    def __eq__(self, other):
+        if isinstance(other, _Signal):
+            other = other._val
+        if isinstance(other, fixbv):
+            assert self._W._fwl == other._W._fwl, \
```
Comment (cfelton) [inline myhdl/_fixbv.py]: Remove assert and implement auto alignment

**Comment by cfelton** [comment]
@qrqiuren can you add [WIP] to the beginning of the title and also name this PR "GSoC fixed-point compiler".

**Comment by qrqiuren** [comment]
@cfelton OK. Done.

**Pull request merged**